test/testcafe/spec/v1/UsingDeviceFlow_spec.js

```diff
@@ -185,4 +210,54 @@ test.requestHooks(requestLogger, legacyDeviceCodeShowLoginMock)('idp discovery a
   await t.expect(deviceCodeActivatePageObject.getPageTitle()).eql('Sign In');
   await t.expect(deviceCodeActivatePageObject.isUserNameFieldVisible()).eql(true);
   await t.expect(deviceCodeActivatePageObject.isPasswordFieldVisible()).eql(false);
+});
+
+test.requestHooks(requestLogger, legacyDeviceCodeShowLoginMockWithoutDeviceFlow)('social login after device activate and redirect', async t => {
+  const deviceCodeActivatePageObject = await setup(t);
+  const identityPage = new IdentityPageObject(t);
+  await rerenderWidget({
+    stateToken: '00-dummy-state-token', //start with 00 to render legacy sign in widget
+    features: {
+      idpDiscovery: true
+    },
+    idps: [
+      {type: 'GOOGLE', id: '0oaaix1twko0jyKik0g1'}
+    ]
+  });
+
+  // submit user code
+  await deviceCodeActivatePageObject.setActivateCodeTextBoxValue('ABCDWXYZ');
+  await deviceCodeActivatePageObject.clickNextButton();
+  await t.expect(deviceCodeActivatePageObject.getPageTitle()).eql('Sign In');
+  await t.expect(deviceCodeActivatePageObject.isUserNameFieldVisible()).eql(true);
+  await t.expect(deviceCodeActivatePageObject.isPasswordFieldVisible()).eql(false);
+  await t.click('.social-auth-google-button');
```

Comment:
probably can create a method in `identityPage` like `identityPage.clickGoogleIdpButton()`

Reply:
I initially tried that but every method on `identityPage` referenced the form, and this was outside of the form, so couldn't find a suitable place for it.